Eggstravaganza/Assets/Scripts/GameManager.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using Unity.Netcode;
+using UnityEngine;
+
+public class GameManager : NetworkBehaviour
+{
+    static GameManager m_Instance;
+    public static GameManager Instance => m_Instance;
+
+    [SerializeField]
+    GameDataScriptableObject GameData;
+    
+    [SerializeField] 
+    UIManager UIManager;
+
+    void Awake()
+    {
+        if (m_Instance != null && m_Instance != this)
+        {
+            Debug.LogError("Multiple GameManagers in scene!");
+            enabled = false; // Destroy() causes NetworkObject issues
+        }
+        else
+        {
+            m_Instance = this;
+        }
+        
+        // DEBUG
+        GameData.Players.Add(0, new PlayerData(0, "Player 0"));
+        GameData.Players.Add(1, new PlayerData(1, "Player 1"));
```

Comment:
minor and possibly silly question, but why do we add these here manually instead of using `RegisterNewPlayer`?

Reply:
good q! i think we'll need to serialize player data upon joining the game (mostly just potentially a custom name and the client ID), but it felt like enough for a separate ticket, so i made #24 for registering upon joining